### PR TITLE
Update puppeteer to version 9

### LINKED ISSUE #6876: Update `puppeteer` to latest version
## Task Description

Need to silence `SharedArrayBuffer` console warnings in E2E tests.

Updating Puppeteer is important because it contains the latest browser version that we want to test against and be compatible with.

### PR DESCRIPTION
**Pull request by spacedmonkey**
### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6876

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]
LGTM 👍 

Just a merge conflict at the moment

**Comment by swissspidy** [comment]
Another merge conflict now (probably because of the percy update) 🙃

**Comment by swissspidy** [comment]
I restarted the e2e tests a few times now, but they keep failing 😞

**Comment by spacedmonkey** [comment]
> I restarted the e2e tests a few times now, but they keep failing 😞

The issue seem to come from upgrade from 9.0.0 to 9.1.0. I locked to down to 9.0.0 and seems to fix the issue. I can't debug this issue locally.

**Comment by spacedmonkey** [comment]
@swissspidy Tests are now passing I had to do the following. 

- Lock puppeteer version to 9.0.0. 
- Ignore a new accessilibity rule. 
- Fixed another new accessilibity rule by adding aria that are invalid.

**Comment by swissspidy** [comment]
Note that now unit tests, integrations tests, and lints are failing

File: babel.config.cjs [hunk 1]
```diff
@@ -49,6 +49,7 @@ module.exports = function (api) {
       'babel-plugin-styled-components',
       ['@babel/plugin-proposal-class-properties', { loose: true }],
       ['@babel/plugin-proposal-private-methods', { loose: true }],
+      ['@babel/plugin-proposal-private-property-in-object', { loose: true }],
```
Comment (swissspidy) [inline babel.config.cjs:52]: Can you explain this change? I don't see how it's related to this.

**Comment by swissspidy** [comment]
It seems like there are inadvertent changes to `package-lock.json` here. Undoing all these changes and then running `npm install` should fix this.

That might help with the failing checks here.

**Review by swissspidy — CHANGES_REQUESTED** [review]
There are some other failing checks now, probably due to lock file changes.

Otherwise ESLint wouldn't have been updated, for example.

This PR should only update Puppeteer, not other packages.

File: babel.config.cjs (hunk 1 above)
Comment (spacedmonkey) [reply babel.config.cjs:52]: One of the dependencies updated and this line was needed.

Comment (swissspidy) [reply babel.config.cjs:52]: As per my other comments, it seems like those dependency updates were accidental

File: packages/e2e-tests/src/config/bootstrap.js [hunk 2]
```diff
@@ -216,6 +216,8 @@ async function runAxeTestsForStoriesEditor() {
       'page-has-heading-one',
       'region',
       'scrollable-region-focusable',
+      'nested-interactive',
+      'aria-allowed-attr',
```
Comment (spacedmonkey) [inline packages/e2e-tests/src/config/bootstrap.js:220]:
```suggestion
      'aria-allowed-attr',
      'link-name',
```

**Pull request closed without merging**